cherab/tools/observers/group/base.py

```diff
@@ -83,420 +78,257 @@ def __getitem__(self, item):
+        for sight_line in self._observers:
             sight_line.observe()
-
-    def _get_same_pipelines(self, item):
-        pipelines = []
-        sight_lines = []
-        for sight_line in self._sight_lines:
-            try:
-                pipelines.append(sight_line.get_pipeline(item))
-            except (ValueError, IndexError):
-                continue
-            else:
-                sight_lines.append(sight_line)
-
-        if len(pipelines) == 0:
-            raise ValueError("Pipeline {} was not found for any sight-line in this {}.".format((item, self.__class__.__name__)))
-
-        pipeline_types = set(type(pipeline) for pipeline in pipelines)
-        if len(pipeline_types) > 1:
-            raise ValueError("Pipelines {} have different types for different sight-lines.".format(item))
-
-        return pipelines, sight_lines
-
-    def plot_total_signal(self, item=0, ax=None):
-        """
-        Plots total (wavelength-integrated) signal for each sight line in the group.
-
-        :param str/int item: The index or name of the pipeline. Default: 0.
-        :param Axes ax: Existing matplotlib axes.
-
-        """
-
-        pipelines, sight_lines = self._get_same_pipelines(item)
-
-        if ax is None:
-            _, ax = plt.subplots(constrained_layout=True)
-
-        signal = []
-        tick_labels = []
-        for pipeline, sight_line in zip(pipelines, sight_lines):
-            if isinstance(pipeline, SpectralPowerPipeline0D):
-                spectrum = Spectrum(pipeline.min_wavelength, pipeline.max_wavelength, pipeline.bins)
-                spectrum.samples = pipeline.samples.mean
-                signal.append(spectrum.total())
-            else:
-                signal.append(pipeline.value.mean)
-
-            if sight_line.name and len(sight_line.name):
-                tick_labels.append(sight_line.name)
-            else:
-                tick_labels.append(self._sight_lines.index(sight_line))
-
-        if isinstance(pipeline, (SpectralRadiancePipeline0D, RadiancePipeline0D)):
-            ylabel = 'Radiance (W/m^2/str)'
-        else:  # SpectralPowerPipeline0D or PowerPipeline0D
-            ylabel = 'Power (W)'
-
-        ax.bar(list(range(len(signal))), signal, tick_label=tick_labels, label=item)
-
-        if isinstance(item, int):
-            # check if pipelines share the same name
-            if len(set(pipeline.name for pipeline in pipelines)) == 1 and pipelines[0].name and len(pipelines[0].name):
-                ax.set_title('{}: {}'.format(self.name, pipelines[0].name))
-            else:
-                # pipelines have different names or name is not set
-                ax.set_title('{}: pipeline {}'.format(self.name, item))
-        elif isinstance(item, str):
-            ax.set_title('{}: {}'.format(self.name, item))
-
-        ax.set_ylabel(ylabel)
-        ax.set_xlabel('Line of sight')
-
-        return ax
-
-    def plot_spectra(self, item=0, in_photons=False, ax=None):
```

Comment:
I think this method is not well suited for the base class as this class can be used with other than spectroscopic pipelines. It seems to me that in this case it would make sense to have a standalone function taking Observer0DGroup or list of pipelines as a parameter.